cleanup of conda related scripts that are no longer needed

shaikh-rashid opened the pull request:
This PR replaces the use of `conda` with `pipenv`. Users will no longer have to install binaries or maintain large conda environment. As part of the change, base install will no longer require the use `make bin-all` or `make init-conda`, both have been replaced by `make init-pipenv`.

Testing should be done by running the following commands:
Remove miniconda and related miniconda config from `~/.bashrc`
`make bin-pyenv`
`which pyenv || exec bash`
`make init-pipenv`
`pipenv shell`
`make init-docker`
`make compose`
`make init-authx`

Reviewers should test on the folllowing platforms:
- [x] Windows
- [x] Linux
- [x] Mac OS X
- [ ] Mac OS X (M1)

Commit:
cleanup of conda related scripts that are no longer needed

Commit:
cleanup of conda related scripts that are no longer needed

Commit:
rebase with branch develop

Commit:
pipenv use instead of conda

Commit:
update docs

Commit:
update docs

daisieh reviewed (changes requested):
If my system doesn't have the current VENV_PYTHON (right now, 3.10), it fails.

```
(candig) daisieh2019:CanDIGv2$ make init-pipenv
echo "    started init-pipenv" >> /Users/daisie/Documents/Work/Sandbox/candig/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
Loading .env environment variables...
Warning: Python 3.10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
make: *** [init-pipenv] Error 1```

shaikh-rashid commented:
@daisieh looks like you have conda still active while using pipenv. I think you will need to deactivate it and try again.

daisieh commented:
Deactivate or uninstall? Because deactivate did not solve the problem.

daisieh commented:
```
(candig) daisieh2019:CanDIGv2$ conda deactivate
(base) daisieh2019:CanDIGv2$ make init-pipenv
echo "    started init-pipenv" >> /Users/daisie/Documents/Work/Sandbox/candig/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
Loading .env environment variables...
Warning: Python 3.10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
make: *** [init-pipenv] Error 1
```

shaikh-rashid commented:
@daisieh you still have the base conda env activated.

daisieh commented:
I had to comment out a bunch of stuff in my .bash_profile to make conda go away, but I am still getting errors:
```
daisieh2019:CanDIGv2$ make init-pipenv
echo "    started init-pipenv" >> /Users/daisie/Documents/Work/Sandbox/candig/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
  WARNING: The script virtualenv-clone is installed in '/Users/daisie/Library/Python/3.9/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script virtualenv is installed in '/Users/daisie/Library/Python/3.9/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts pipenv and pipenv-resolver are installed in '/Users/daisie/Library/Python/3.9/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
WARNING: There was an error checking the latest version of pip.
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
Loading .env environment variables...
Warning: Python 3.10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
make: *** [init-pipenv] Error 1
daisieh2019:CanDIGv2$ 
```

shaikh-rashid commented:
@daisieh the warning indicates that you do not have your python virtualenv and host python set up. You will need to resolve the issues regarding PATH and try again.

daisieh commented:
But isn't the idea of this target supposed to be to make the setup automatic?

kcranston commented on Makefile:
```diff
@@ -34,41 +31,6 @@ mkdir:
-# make bin-conda
-
-#<<<
-bin-conda: mkdir
```
this target needs to be replicated in init-pipenv, else you don't have a `tmp` dir for the logfile (among other paths):
`bash: /Users/karen/Documents/candig/repos/CanDIGv2/tmp/progress.txt: No such file or directory`

shaikh-rashid replied:
Fixed with latest push

kcranston commented:
After resolving the lack of `tmp` dir by running `make mkdir` manually, I get these errors:

```
(base) $  CanDIGv2 git:(conda-venv) make init-pipenv
echo "    started init-pipenv" >> /Users/karen/Documents/candig/repos/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
  WARNING: The script virtualenv-clone is installed in '/Users/karen/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script virtualenv is installed in '/Users/karen/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts pipenv and pipenv-resolver are installed in '/Users/karen/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
bash: pipenv: command not found
make: *** [init-pipenv] Error 127
```

kcranston commented:
What does `pipenv` give us that the default `venv` does not?

If we are going to use pipenv instead of venv (and I am not yet convinced), then we should also use a Pipfile isntead of requirements.txt.

Commit:
add an activate.sh for creating pipenv shell

Commit:
add an activate.sh for creating pipenv shell

Commit:
update doc for pipenv

Commit:
update doc for pipenv

Commit:
moved init-pipenv steps into activate.sh

Commit:
moved init-pipenv steps into activate.sh

Commit:
need mkdir for init-docker

Commit:
need mkdir for init-docker

Commit:
activate.sh now selfcontained init

Commit:
activate.sh now selfcontained init

Commit:
update docs

Commit:
update docs

daisieh commented:
Is there a reason to *replace* conda with either pipenv or venv, or can we just have all of them as options?

daisieh committed:
Move settings.py here (#190)

* settings.py is a convenient script

* add minio vars

* move settings.py here, not in ingest

OrdiNeu committed:
DIG-996: Automate /etc/hosts step during init-authx (#188)

* DIG-996: Automate /etc/hosts step during init-authx

* DIG-996: Vault startup commands fix for MacOS

* Fix documentation on hosts editing

* DIG-996 doc

* DIG-996: Move the hosts setup to outside of the vault step, into its own step as part of init-docker

Also skip the step if it is not required, and warn the user if multiple
IP addresses were detected

* Swap tab with whitespace

* [Misc] Fix bug where echo was removing newlines

* Move init-hosts-file into its own step

Commit:
update to revert pipenv changes

Commit:
update to revert pipenv changes

Commit:
update branch with changes from develop (rebase)

Commit:
added pyenv for python support

Commit:
minor doc changes

daisieh reviewed (approved):
Except for the weird edge case that I ran into that seems to have had nothing to do with this branch or changes (and resolved itself with a few clean/inits), this works fine and much faster than conda. Ran through init-authx successfully and was able to clean successfully as well.

kcranston commented:
The `bin-pyenv` step writes some configuration to `.bashrc`. If you are using a different shell (e.g. zsh is default on newer macs), later steps fail because this needs to be replicated in `.zshrc`. So, this PR should include documentation on a workaround for zsh (either instructions for adding the config to `.zshrc` and / or adding the filename of the login script as a variable).

shaikh-rashid committed:
require python-dev

OrdiNeu reviewed (approved):
Works for me, tested a while ago

CourtneyGosselin reviewed (approved):
LGTM
